src/dune_lang/package_constraint.mli

```diff
@@ -0,0 +1,47 @@
+open! Stdune
+
+module Op : sig
+  type t =
+    | Eq
+    | Gte
+    | Lte
+    | Gt
+    | Lt
+    | Neq
+end
+
+module Variable : sig
+  type t = { name : string }
+end
+
+module Value : sig
+  type t =
+    | String_literal of string
+    | Variable of Variable.t
+end
+
+(** A constraint for use in package definitions, equivalent to the filter
+    language used in opam package definitions. All expressions in this language
+    are booleans. String literal values can only appear as the argument to an
+    operator which returns a boolean. Variables are dynamically typed and can
+    represent strings and booleans. If a variable appears in a position where a
+    boolean is expected it will be assumed to represent a boolean. *)
+type t =
```

Comment:
Dumb question: Is our blang really that different from this?

Reply:
Good question. I don't know off the top of my head. Of course they both have similar functions though the package constraint language has the added requirement of being translatable to the opam filter language. At some point soon I'll be looking at actually evaluating conditions (currently we only use these to generate opam files) and at that point I'll see whether it makes sense to just replace this with blang.